mbuild/utils/conversion.py

```diff
@@ -2,26 +2,98 @@
 import numpy as np
 
 
-def RB_to_OPLS(c0, c1, c2, c3, c4, c5):
-    """Convert Ryckaert-Bellemans type dihedrals to OPLS type.
+def RB_to_OPLS(c0, c1, c2, c3, c4, c5, error_tol=1e-4):
+    r"""Convert Ryckaert-Bellemans type dihedrals to OPLS type.
+
+    .. math::
+    RB_torsions &= c0 + c1*cos(psi) + c2*cos(psi)^2 + c3*cos(psi)^3 + \\
+                &= c4*cos(psi)^4 + c5*cos(5*psi)^5
+
+    .. math::
+    OPLS_torsions &= f1/2*(1+cos(t)) + f2/2*(1-cos(2*t)) + \\
+                  &= f3/2*(1+cos(3*t)) + f4/2(1-cos(4*t))
+
+    where :math:`psi = t - pi = t - 180 degrees`
 
     Parameters
     ----------
     c0, c1, c2, c3, c4, c5 : Ryckaert-Belleman coefficients (in kcal/mol)
+    error_tol : float, default=1e-4
+        The acceptable absolute tolerance between the RB to OPLS conversion.
+        Any value entered is converted to an absolute value.
 
     Returns
     -------
     opls_coeffs : np.array, shape=(4,)
         Array containing the OPLS dihedrals coeffs f1, f2, f3, and f4
-        (in kcal/mol)
+        (in kcal/mol).
+        NOTE: fO IS NOT IN THE OPLS DIHEDRAL EQUATION AND IS ONLY USED
+        TO TEST IF THIS FUNCTION CAN BE UTILIZED. ONE OF THE
+        CONDITIONS OF CONVERSION IS f0 MUST BE ZERO (f0=0).
+
+    Notes
+    -----
+    c5 must equal zero, or this conversion is not possible.
+
+    (c0 + c1 + c2 + c3 + c4 + c5) must equal zero, or this conversion is not possible.
     """
-    f1 = (-1.5 * c3) - (2 * c1)
-    f2 = c0 + c1 + c3
-    f3 = -0.5 * c3
-    f4 = -0.25 * c4
+    if isinstance(error_tol, float):
```

Comment:
Lets just coerce this variable into a float, and that way we dont have to worry too much about this

```suggestion
    error_tol = float(error_tol)
```

Reply:
I put it this way so if they enter a list, dict, or string it fails.  Additionally,

Reply:
Maybe im misunderstanding, but does typecasting to a float not give you this behavior?

```python3
>>> a = dict()
>>> a
{}
>>> b = set()
>>> float(b)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: float() argument must be a string or a number, not 'set'
>>> float(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: float() argument must be a string or a number, not 'dict'
```

Reply:
Changed to:

    try:
        error_tol = abs(float(error_tol))
    except:
        raise TypeError("ERROR: The error_tol variable must be a float.")